### ISSUE TITLE
mu4e combines declared folders when using context

### ISSUE THREAD

**Issue by averter** [djcb/mu#1172]
## Expected or desired behavior
mu4e-context should separate mail from different accounts Inbox, Trash, Sent and Draft folders

## Actual behavior
When I jump into Inbox, Trash, Sent or Draft folders I see emails from both my accounts mixed together, although they are in two separate directory structures which I have declared.

## Versions of mu, mu4e/emacs, operating system etc.
mu is 0.9.11 and mu4e is 0.9.12-3. I am on Linux mint cinnamon 18.3 64 bit.

## Any other detail
Here is my mu4e config  [file](https://gist.github.com/averter/021989b1566fc53763952441df1f2391). My emails have been synchronized into two directories user@gmail.com and user@warwick.ac.uk under a ~/Mail/ root directory.

I also have another error when I start mu4e that could be related. Specifically: `error in process filter: mu4e-error: [mu4e] mu server has version 0.9.11, but we need 0.9.17`

**Comment by notmgsk** [comment]
The purpose of `mu4e` contexts is to change settings depending on the context. If you want to configure `mu4e` to show only email associated with the context email address, then take a look at the `enter-func` property of the `mu4e-context` structure https://www.djcbsoftware.nl/code/mu/mu4e/What-are-contexts.html#What-are-contexts

**Comment by averter** [comment]
Maybe I am not using the correct tool for the job. All I want is to replace thunderbird with mu4e having a similar setup (i.e. multiple separate accounts). I am not an experienced elisp user to be honest. What would be the best course of action to attain what I want? And if it is with the `enter-func` property what should I write on it? Thanks

**Comment by notmgsk** [comment]
One solution is to add bookmarks to the `mu4e-bookmarks` list. A bookmark is essentially a search query with a name and a keybinding. For example, you might have a bookmark to show only unread mail, and indeed mu4e has this by default bound to the keys `bu`.

You might create a bookmark that shows emails that are sent to a particular work email address (either directly or cc):

```
(add-to-list 'mu4e-bookmarks
    (make-mu4e-bookmark
    :name "Work emails"
    :query "to:myworkemail@company.com"
    :key ?s))
```
The keys `bs` will then show those emails.

**Comment by averter** [comment]
Thank you @notmgsk . I will definitely start using bookmarks and shortcuts now that I was able to set up everything. For anyone who is facing problems at setting multiple accounts in mu4e after having used one account like me, here is what I did:
1- One has to run `mu index --rebuild --maildir=~/Mail` This command did the trick. Essentially one should not do the indexing separately for each of the directories below the main directory. Moreover the `rebuild` option that rebuilds the database from scratch is essential. Hope this helps other people.
